### PR TITLE
Added support for MSC3881

### PR DESCRIPTION
**Pull request by gileluard**
needed for [#6787](https://github.com/vector-im/element-ios/issues/6787)

### CODE REVIEW

File: MatrixSDK/MXRestClient.m [hunk 1]
```diff
@@ -3200,7 +3228,7 @@ - (MXHTTPOperation*)roomSummaryWith:(NSString*)roomIdOrAlias
                             success:(void (^)(MXPublicRoom *room))success
                             failure:(void (^)(NSError *error))failure
 {
-    NSMutableString *path = [NSMutableString stringWithFormat:@"%@/im.nheko.summary/rooms/%@/summary", kMXAPIPrefixPathUnstable, roomIdOrAlias];
+    NSMutableString *path = [NSMutableString stringWithFormat:@"%@/im.nheko.summary/rooms/%@/summary", kMXAPIPrefixPathUnstable, [MXTools encodeURIComponent:roomIdOrAlias]];
```
Comment (pixlwave) [inline MatrixSDK/MXRestClient.m:3231]: Looks sensible, but not sure I understand how this change relates to the MSC.

File: MatrixSDK/JSONModels/MXMatrixVersions.m [hunk 2]
```diff
@@ -42,6 +42,8 @@
 
 //  Unstable features
 static NSString* const kJSONKeyMSC3440 = @"org.matrix.msc3440.stable";
+static NSString* const kJSONKeyMSC3881 = @"org.matrix.msc3881";
+static NSString* const kJSONKeyMSC3881Stable = @"org.matrix.msc3881.stable";
```
Comment (pixlwave) [inline MatrixSDK/JSONModels/MXMatrixVersions.m:46]: Given `kJSONKeyMSC3440` is stable, would it make more sense to do this like so?
```suggestion
static NSString* const kJSONKeyMSC3881Unstable = @"org.matrix.msc3881";
static NSString* const kJSONKeyMSC3881 = @"org.matrix.msc3881.stable";
```

File: MatrixSDK/Contrib/Swift/JSONModels/MXJSONModels.swift [hunk 3]
```diff
@@ -56,6 +56,18 @@ public enum MXLoginFlowType: Equatable, Hashable {
 public enum MXPusherKind: Equatable, Hashable {
     case http, none, custom(String)
     
+    public static func from(value: String?) -> MXPusherKind {
```
Comment (pixlwave) [inline MatrixSDK/Contrib/Swift/JSONModels/MXJSONModels.swift:59]: Unimportant, but wondering if this could be an `init`?

**Review by pixlwave — APPROVED** [review]
LGTM 😎

File: MatrixSDK/MXRestClient.m (hunk 1 above)
Comment (gileluard) [reply MatrixSDK/MXRestClient.m:3231]: Indeed. I've just added this code to fix some eventual issues. Do you want me to revert it?

Comment (gileluard) [reply MatrixSDK/MXRestClient.m:3231]: This could definitely happen to me :D

**Pull request merged**